src/sagemaker/model.py

```diff
@@ -549,7 +549,10 @@ def register(
             model_package_group_name = utils.base_name_from_image(
                 self.image_uri, default_base_name=ModelPackage.__name__
             )
-        if model_package_group_name is not None:
+        if (
+            model_package_group_name is not None
+            and model_type is not JumpStartModelType.PROPRIETARY
```

Comment:
why adding this new condition? We do not need this new condition because we want for any register function without model_package_group_name to auto populate group_name and not just for JumpStartModelType.PROPRIETARY

Reply:
for the proprietary model, it uses source URI and has MPG name passing from JumpStartModel class. causing container def construction failure inside Model register.
so i added a condition inside model register to not to construct it. but the mpg name setup and assignments all are placed inside jumpstart class